pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -134,6 +134,9 @@ func addDateHistogramAgg(aggBuilder es.AggBuilder, bucketAgg *BucketAgg, timeFro
 			a.Interval = "$__interval"
 		}
 
+		if bucketAgg.Settings.Get("offset").MustString("") != "" {
```

Comment:
I would suggest using similar logic as for `missing` below, i.e.

```go
if offset, err := bucketAgg.Settings.Get("offset").String(); err == nil {
  a.Offset = offset
}
```

Alternatively assigning `bucketAgg.Settings.Get("offset").MustString("")` to a variable so that you don't need to do the same lookup twice.

Reply:
Committed changes